lib/chef/provider/service/systemd.rb

```diff
@@ -173,25 +198,29 @@ def unmask_service
   end
 
   def is_active?
-    options, args = get_systemctl_options_args
-    shell_out(systemctl_path, args, "is-active", new_resource.service_name, "--quiet", **options).exitstatus == 0
+    status = systemd_service_status
+    # Note: "activating" is not active (as with type=notify or oneshots)
+    status["ActiveState"] == "active"
   end
 
   def is_enabled?
-    options, args = get_systemctl_options_args
-    shell_out(systemctl_path, args, "is-enabled", new_resource.service_name, "--quiet", **options).exitstatus == 0
+    status = systemd_service_status
+    # See https://github.com/systemd/systemd/blob/master/src/systemctl/systemctl-is-enabled.c
+    # Note: enabled-runtime is excluded because this is volatile, and the state of enabled-runtime
+    # specifically means that the service is not enabled
+    %w{enabled static generated alias indirect}.include?(status["UnitFileState"])
```

Comment:
```suggestion
    # See https://github.com/systemd/systemd/blob/master/src/systemctl/systemctl-is-enabled.c
    # Note: enabled-runtime is excluded because this is volatile, and the state of enabled-runtime
    # specifically means that the service is not enabled
    %w{enabled static generated alias indirect}.include?(systemd_service_status["UnitFileState"])
```